astropy/modeling/functional_models.py

```diff
@@ -714,20 +749,33 @@ class Sine1D(Fittable1DModel):
     @staticmethod
     def evaluate(x, amplitude, frequency, phase):
         """One dimensional Sine model function"""
-
-        return amplitude * np.sin(2 * np.pi * frequency * x + 2 * np.pi * phase)
+        argument = TWOPI * (frequency * x + phase)
+        if isinstance(argument, Quantity):
```

Comment:
Ah yes - I admit it's a little confusing - the reason for this is because if frequency and x are quantities, argument will end up being dimensionless because TWOPI doesn't have a unit (even though it should be radians). But then np.sin(dimensionless) crashes.

Another option would be to do ``argument = argument.value`` which would work too. I might do that instead since that will work too, and will put a comment.